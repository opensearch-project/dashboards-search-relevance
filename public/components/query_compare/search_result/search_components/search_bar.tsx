/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton, EuiFieldSearch, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
import React from 'react';

interface SearchBarProps {
  searchBarValue: string;
  setSearchBarValue: React.Dispatch<React.SetStateAction<string>>;
  onClickSearch: () => void;
}

export const SearchInputBar = ({
  searchBarValue,
  setSearchBarValue,
  onClickSearch,
}: SearchBarProps) => {
  return (
    <>
      <EuiSpacer size="m" />
      <EuiFlexGroup>
        <EuiFlexItem grow={true}>
          <EuiFieldSearch
            id="searchRelevance-searchBar"
            fullWidth={true}
            placeholder="Search"
            value={searchBarValue}
            onChange={(e) => setSearchBarValue(e.target.value)}
            isClearable={true}
            onSearch={onClickSearch}
            aria-label="Enter your Search query"
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButton fill onClick={onClickSearch} aria-label="searchRelevance-searchButton">
            Search
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    </>
  );
};
