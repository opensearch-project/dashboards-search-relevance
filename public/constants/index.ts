/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

export enum ServiceEndpoints {
  GetIndexes = '/api/relevancy/search/indexes',
  GetSearchResults = '/api/relevancy/search',
}
