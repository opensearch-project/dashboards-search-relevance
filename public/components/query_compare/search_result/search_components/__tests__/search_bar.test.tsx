/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { SearchInputBar } from '../search_bar';
import { TEST_SEARCH_TEXT } from '../../../../../../test/constants';

describe('Search bar component', () => {
  configure({ adapter: new Adapter() });

  it('Renders search bar component', async () => {
    const wrapper = mount(
      <SearchInputBar
        searchBarValue={TEST_SEARCH_TEXT}
        setSearchBarValue={() => {}}
        onClickSearch={() => {}}
      />
    );

    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
