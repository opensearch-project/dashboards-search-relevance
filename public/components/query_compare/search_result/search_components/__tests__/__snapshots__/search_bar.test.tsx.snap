// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component Renders search bar component 1`] = `
<SearchInputBar
  onClickSearch={[Function]}
  searchBarValue="basic"
  setSearchBarValue={[Function]}
>
  <EuiSpacer
    size="m"
  >
    <div
      className="euiSpacer euiSpacer--m"
    />
  </EuiSpacer>
  <EuiFlexGroup>
    <div
      className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
    >
      <EuiFlexItem
        grow={true}
      >
        <div
          className="euiFlexItem"
        >
          <EuiFieldSearch
            aria-label="Enter your Search query"
            compressed={false}
            fullWidth={true}
            id="searchRelevance-searchBar"
            incremental={false}
            isClearable={true}
            isLoading={false}
            onChange={[Function]}
            onSearch={[Function]}
            placeholder="Search"
            value="basic"
          >
            <EuiFormControlLayout
              clear={
                Object {
                  "onClick": [Function],
                }
              }
              compressed={false}
              fullWidth={true}
              icon="search"
              isLoading={false}
            >
              <div
                className="euiFormControlLayout euiFormControlLayout--fullWidth"
              >
                <div
                  className="euiFormControlLayout__childrenWrapper"
                >
                  <EuiValidatableControl>
                    <input
                      aria-label="Enter your Search query"
                      className="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
                      id="searchRelevance-searchBar"
                      onChange={[Function]}
                      onKeyUp={[Function]}
                      placeholder="Search"
                      type="search"
                      value="basic"
                    />
                  </EuiValidatableControl>
                  <EuiFormControlLayoutIcons
                    clear={
                      Object {
                        "onClick": [Function],
                      }
                    }
                    compressed={false}
                    icon="search"
                    isLoading={false}
                  >
                    <div
                      className="euiFormControlLayoutIcons"
                    >
                      <EuiFormControlLayoutCustomIcon
                        size="m"
                        type="search"
                      >
                        <span
                          className="euiFormControlLayoutCustomIcon"
                        >
                          <EuiIcon
                            aria-hidden="true"
                            className="euiFormControlLayoutCustomIcon__icon"
                            size="m"
                            type="search"
                          >
                            <EuiIconEmpty
                              aria-hidden={true}
                              className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                              focusable="false"
                              role="img"
                              style={null}
                            >
                              <svg
                                aria-hidden={true}
                                className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
                                focusable="false"
                                height={16}
                                role="img"
                                style={null}
                                viewBox="0 0 16 16"
                                width={16}
                                xmlns="http://www.w3.org/2000/svg"
                              />
                            </EuiIconEmpty>
                          </EuiIcon>
                        </span>
                      </EuiFormControlLayoutCustomIcon>
                    </div>
                    <div
                      className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                    >
                      <EuiFormControlLayoutClearButton
                        onClick={[Function]}
                        size="m"
                      >
                        <EuiI18n
                          default="Clear input"
                          token="euiFormControlLayoutClearButton.label"
                        >
                          <button
                            aria-label="Clear input"
                            className="euiFormControlLayoutClearButton"
                            onClick={[Function]}
                            type="button"
                          >
                            <EuiIcon
                              className="euiFormControlLayoutClearButton__icon"
                              type="cross"
                            >
                              <EuiIconEmpty
                                aria-hidden={true}
                                className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutClearButton__icon"
                                focusable="false"
                                role="img"
                                style={null}
                              >
                                <svg
                                  aria-hidden={true}
                                  className="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutClearButton__icon"
                                  focusable="false"
                                  height={16}
                                  role="img"
                                  style={null}
                                  viewBox="0 0 16 16"
                                  width={16}
                                  xmlns="http://www.w3.org/2000/svg"
                                />
                              </EuiIconEmpty>
                            </EuiIcon>
                          </button>
                        </EuiI18n>
                      </EuiFormControlLayoutClearButton>
                    </div>
                  </EuiFormControlLayoutIcons>
                </div>
              </div>
            </EuiFormControlLayout>
          </EuiFieldSearch>
        </div>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <div
          className="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <EuiButton
            aria-label="searchRelevance-searchButton"
            fill={true}
            onClick={[Function]}
          >
            <EuiButtonDisplay
              aria-label="searchRelevance-searchButton"
              baseClassName="euiButton"
              disabled={false}
              element="button"
              fill={true}
              isDisabled={false}
              onClick={[Function]}
              type="button"
            >
              <button
                aria-label="searchRelevance-searchButton"
                className="euiButton euiButton--primary euiButton--fill"
                disabled={false}
                onClick={[Function]}
                style={
                  Object {
                    "minWidth": undefined,
                  }
                }
                type="button"
              >
                <EuiButtonContent
                  className="euiButton__content"
                  iconSide="left"
                  textProps={
                    Object {
                      "className": "euiButton__text",
                    }
                  }
                >
                  <span
                    className="euiButtonContent euiButton__content"
                  >
                    <span
                      className="euiButton__text"
                    >
                      Search
                    </span>
                  </span>
                </EuiButtonContent>
              </button>
            </EuiButtonDisplay>
          </EuiButton>
        </div>
      </EuiFlexItem>
    </div>
  </EuiFlexGroup>
</SearchInputBar>
`;
